code/app/scripts/collatz_client_server.xml

```diff
@@ -0,0 +1,58 @@
+<application>
+    <name> collatz_client_server </name>
+
+    <module>
+        <name> collatz-server </name>
+        <node> localhost </node>
+    </module>
+
+    <module>
+        <name> collatz-client </name>
+        <node> localhost </node>
+        <parameters>--id a</parameters>
+    </module>
+
+    <module>
+        <name> collatz-client </name>
+        <node> localhost </node>
+        <parameters>--id b</parameters>
+    </module>
+
+    <module>
+        <name> collatz-client </name>
+        <node> localhost </node>
+        <parameters>--id c</parameters>
+    </module>
+
+    <module>
+        <name> collatz-client </name>
+        <node> localhost </node>
+        <parameters>--id d</parameters>
+    </module>
+
+    <connection>
+        <from>/Client_a</from>
+        <to>/Server</to>
+        <protocol>tcp</protocol>
+    </connection>
+
+
+    <connection>
+        <from>/Client_b</from>
+        <to>/Server</to>
+        <protocol>tcp</protocol>
```

Comment:
[`TCP`][1] is a **connection oriented protocol**, meaning that when for any reason a packet gets lost it is transmitted again. Thus, it turns to be particularly useful for asynchronous channel (e.g. **rpc**) where it's crucial not to loose information (e.g. the reply).

[`UDP`][2] is instead a **connectionless protocol** that does not provide packets retransmission. It is then designed for synchronous communication (i.e. **streaming**), where you don't mind to loose one packet since you receive fresh data continuously.

For a related discussion, see also https://github.com/robotology/QA/issues/139.

[1]: https://en.wikipedia.org/wiki/Transmission_Control_Protocol
[2]: https://en.wikipedia.org/wiki/User_Datagram_Protocol